source/http_stream.c

```diff
@@ -163,16 +179,26 @@ static void s_on_complete_call(napi_env env, napi_value on_complete, void *conte
     struct http_stream_binding *binding = context;
     struct on_complete_args *args = user_data;
 
-    if (env) {
-        napi_value params[1];
-        const size_t num_params = AWS_ARRAY_SIZE(params);
-
-        AWS_NAPI_ENSURE(env, napi_create_int32(env, args->error_code, &params[0]));
-        AWS_NAPI_ENSURE(
-            env,
-            aws_napi_dispatch_threadsafe_function(env, binding->on_complete, NULL, on_complete, num_params, params));
+    if (binding->pending_length) {
```

Comment:
Do we still need this machinery if we have proven that the queued TS functions are in fact in order?

Reply:
What I found is that the complete callback **did** happen before some of body callbacks, even if from the C land, the complete will be the last one to be invoked. But, all the body callbacks happen in the right order. 
I still don't know about how nodejs dealing with it. 
But some guessing here: 
From their [samples](https://nodejs.github.io/node-addon-examples/special-topics/thread-safe-functions/ ), the TS functions are invoked as the same function in nodejs, which ensures the order. But, the different functions may be out of order.